szkolenie/szkolenie/Lesson4.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Interactions;
+using OpenQA.Selenium.Support.UI;
+
+namespace szkolenie
+{
+    class Lesson4 : Fixtures
+    {
+        [TestCase("https://www.seleniumeasy.com/test/dynamic-data-loading-demo.html", "false")]
+        public void ExplicitWaitImage(string URL, string startup_popup)
+        {
+            // arrange
+            var newUser = driver.FindElement(By.ClassName("btn-default"));
+            var loader = driver.FindElement(By.Id("loading"));
+            
+            // act
+            newUser.Click();
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
+            wait.Until(d => loader.Text.Contains("First Name"));
+
+            // assert
+            Assert.That(loader.Text.Contains("First Name"));
+            Assert.That(loader.Text.Contains("Last Name"));
+        }
+
+        [TestCase("https://www.seleniumeasy.com/test/jquery-download-progress-bar-demo.html", "false")]
+        public void ExplicitWaitPorgressBar(string URL, string startup_popup)
+        {
+            // arrange
+            var start = driver.FindElement(By.Id("downloadButton"));
+            var button = driver.FindElement(By.ClassName("ui-dialog-buttonset"));
+
+            // act
+            start.Click();
+            //var progressBar = driver.FindElement(By.Id("progressbar")).GetAttribute("aria-valuenow");
```

Comment:
I debugged this, and just after clicking start button, progrressbar element appears but without 'aria-valuenow' attribute. Here is screenshot with paused javascript on the page in this moment. Solution is to add here some wait until this attribute is present and then dynamically wait until it has value of 100.
![image](https://user-images.githubusercontent.com/40823115/112608244-659cc980-8e1a-11eb-9589-bf6199d3db0a.png)

Reply:
thanks, I will test it next time.